contributor-docs/best_practices/dependency_injection.md

```diff
@@ -0,0 +1,247 @@
+---
+parent: Best Practices
+---
+
+# Dependency Injection
+
+While [Firebase Components]({{ site.baseurl }}{% link components/components.md %}) provides basic
+Dependency Injection capabilities for interop between Firebase SDKs, it's not ideal as a general purpose
+DI framework or a couple of reasons, to name a few:
+
+* It's verbose, i.e. requires manually specifying dependencies and constructing instances of components in Component
+  definitions.
+* It has a runtime cost, i.e. initialization time is linear in the number of Components present in the graph
+
+As a result using [Firebase Components]({{ site.baseurl }}{% link components/components.md %}) is appropriate only
+for inter-SDK injection and scoping instances per `FirebaseApp`.
+
+On the other hand, manually instantiating SDKs is often tedious, errorprone, and often leads to code smells
+that make code less testable and coulpes it to implementation rather than the interface. For more context see
+[Dependency Injection](https://en.wikipedia.org/wiki/Dependency_injection) and  [Motivation](https://github.com/google/guice/wiki/Motivation).
+
+{: .important }
+It's recommended to use [Dagger](https://dagger.dev) for internal dependency injection within the SDKs and
+[Components]({{ site.baseurl }}{% link components/components.md %}) to inject inter-sdk dependencies that are available only at
+runtime into the [Dagger Graph](https://dagger.dev/dev-guide/#building-the-graph) via
+[builder setters](https://dagger.dev/dev-guide/#binding-instances) or [factory arguments](https://dagger.dev/api/latest/dagger/Component.Factory.html).
+
+See: [Dagger docs](https://dagger.dev)
+See: [Dagger tutorial](https://dagger.dev/tutorial/)
+
+{: .highlight }
+While Hilt is the recommended way to use dagger in Android `applications, it's not suitable for SDK/library development.
+
+## How to get started
+
+Since [Dagger](https://dagger.dev) does not strictly follow semver and requires the dagger-compiler version to match with the dagger library version,
```

Comment:
```suggestion
Since [Dagger](https://dagger.dev) does not strictly follow semver and requires the dagger-compiler version to match the dagger library version,
```